buildSrc/src/main/java/com/uber/okbuck/core/model/jvm/JvmTarget.java

```diff
@@ -281,6 +285,46 @@ public static String javaVersion(JavaVersion version) {
       return ImmutableList.of();
     }
     ImmutableList.Builder<String> optionBuilder = ImmutableList.builder();
+
+    // Add all compiler arguments
+    String compileTaskName = isTest ? "compileTestKotlin" : "compileKotlin";
```

Comment:
Is there no public API/static constants for these task names? Hardcoding can lead to breakages in future

Reply:
This approach also won't work for kotlin Android targets

Reply:
No public api/constants. Why won't this work for Android targets?

Reply:
Android targets don't have the same task names as they are variant aware

Reply:
Good point. I can do something like `tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {`, but how can I connect those back to the specific variant at hand? Or would it just be basic String matching?

Reply:
alternatively - just grab the first one and only support configurations that apply to all

Reply:
I tend to think people use flavors and buildTypes for feature specific things (changing a hardcoded string value, changing themes, etc...). Could not think of a use case where anyone would want to consistently compile differently depending on that, so I think it's valid to only support configurations that apply to all.

Reply:
Agreed, just updated to take the first matching one in 245e32f

Reply:
Did keep the test vs non-test distinction though. Ultimately it's a dynamic naming strategy though

```kotlin
    override val compileKotlinTaskName: String
        get() = lowerCamelCaseName(
            "compile",
            compilationName.takeIf { it != KotlinCompilation.MAIN_COMPILATION_NAME },
            "Kotlin",
            target.targetName
        )
```